packages/state-transition/src/cache/epochTransitionCache.ts

```diff
@@ -376,7 +380,10 @@ export function beforeProcessEpoch(
   for (let i = 0; i < nextEpochShufflingActiveIndicesLength; i++) {
     nextShufflingActiveIndices[i] = nextEpochShufflingActiveValidatorIndices[i];
   }
-  state.epochCtx.shufflingCache?.build(epochAfterNext, nextShufflingDecisionRoot, state, nextShufflingActiveIndices);
+
+  if (opts?.asyncShufflingCalculation) {
+    state.epochCtx.shufflingCache?.build(epochAfterNext, nextShufflingDecisionRoot, state, nextShufflingActiveIndices);
+  }
```

Comment:
`beforeProcessEpoch` is also called from `computeUnrealizedCheckpoints` which I was not aware when we implemented this feature

protect this call by this new flag fixes it

Reply:
Yep. It should.  Need to pass `true` for it to run async so all other calls in the codebase will default to undefined/falsy and running synchronously